packages/common/src/hooks/usePreventExit.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+
+/**
+ * Hook that prevents the user from leaving the page without confirmation.
+ * @example
+ * ```tsx
+ * import { usePreventExit } from '@monkvision/common';
+ *
+ * function MyComponent() {
+ *  usePreventExit();
+ *  return <div>My Component</div>;
+ * }
+ * ```
+ */
+export function usePreventExit() {
+  useEffect(() => {
+    window.onbeforeunload = () => {
+      return 'Are you sure you want to leave?';
```

Comment:
Instead of displaying a hard coded message :

- Make it so that the component accepts a single prop called `confirmationMessage` which is an optional string (default value : 'Are you sure you want to leave?')
- Then, in the `PhotoCapture`, pass a translated message like this : 

```tsx
usePreventExit(t('exit-message'));
```

Reply:
actually whatever messages we have on alert it always says "you have some unsaved changes" not out messages

Reply:
Okay so what is the point of having this hardcoded string? Can't we just remove it?

Reply:
It should just have some truthy value to show some of the alerts. I have seen mentioned somewhere like on legacy browsers it can will show but now a days modern browsers it's not showing. Probably due to some vulnerability.

Reply:
https://stackoverflow.com/questions/45088861/whats-the-point-of-beforeunload-returnvalue-if-the-message-does-not-get-set